test/parallel/test-net-socket-write-error.js

```diff
@@ -13,7 +13,7 @@ function connectToServer() {
                           type: TypeError
                         });
 
-    client.end();
+    client.destroy();
   })
-  .on('end', () => server.close());
+  .on('close', () => server.close());
```

Comment:
Note here. Since the client is errored the `end()` doesn't emit `finish` which changes so that the socket is not destroyed (since net.socket doesn't have `autoDestroy`).